test/e2e/install_test.go

```diff
@@ -32,237 +34,297 @@ const (
-				operator.Spec.Version = "2.0.0"
-				Expect(c.Update(ctx, operator)).To(Succeed())
-				By("eventually reporting an unsatisfiable resolution")
-				Eventually(func(g Gomega) {
-					g.Expect(c.Get(ctx, types.NamespacedName{Name: operator.Name}, operator)).To(Succeed())
-					cond := apimeta.FindStatusCondition(operator.Status.Conditions, operatorv1alpha1.TypeResolved)
-					g.Expect(cond).ToNot(BeNil())
-					g.Expect(cond.Reason).To(Equal(operatorv1alpha1.ReasonResolutionFailed))
-					g.Expect(cond.Message).To(ContainSubstring("constraints not satisfiable"))
-					g.Expect(cond.Message).To(MatchRegexp("installed package prometheus requires at least one of test-catalog-prometheus-prometheus-operator.1.2.0, test-catalog-prometheus-prometheus-operator.1.0.1, test-catalog-prometheus-prometheus-operator.1.0.0$"))
-					g.Expect(operator.Status.ResolvedBundleResource).To(BeEmpty())
-				}).Should(Succeed())
-			})
-
-			It("does allow to upgrade the Operator to any of the successor versions within non-zero major version", func() {
-				By("updating the Operator resource by skipping versions")
-				// Test catalog has versions between the initial version and new version
-				operator.Spec.Version = "1.2.0"
-				Expect(c.Update(ctx, operator)).To(Succeed())
-				By("eventually reporting a successful resolution and bundle path")
-				Eventually(func(g Gomega) {
-					g.Expect(c.Get(ctx, types.NamespacedName{Name: operator.Name}, operator)).To(Succeed())
-					cond := apimeta.FindStatusCondition(operator.Status.Conditions, operatorv1alpha1.TypeResolved)
-					g.Expect(cond).ToNot(BeNil())
-					g.Expect(cond.Reason).To(Equal(operatorv1alpha1.ReasonSuccess))
-					g.Expect(cond.Message).To(ContainSubstring("resolved to"))
-					g.Expect(operator.Status.ResolvedBundleResource).To(Equal("localhost/testdata/bundles/registry-v1/prometheus-operator:v1.2.0"))
-				}).Should(Succeed())
-			})
-		})
-
-		AfterEach(func() {
-			if basePath := env.GetString("ARTIFACT_PATH", ""); basePath != "" {
-				// get all the artifacts from the test run and save them to the artifact path
-				getArtifactsOutput(ctx, basePath)
-			}
-			Expect(c.Delete(ctx, operator)).To(Succeed())
-			Eventually(func(g Gomega) {
-				err := c.Get(ctx, types.NamespacedName{Name: operator.Name}, &operatorv1alpha1.Operator{})
-				g.Expect(errors.IsNotFound(err)).To(BeTrue())
-			}).Should(Succeed())
-		})
+func testInit(t *testing.T) (*operatorv1alpha1.Operator, string, *catalogd.Catalog) {
+	var err error
+	operatorCatalog, err := createTestCatalog(context.Background(), testCatalogName, os.Getenv(testCatalogRefEnvVar))
+	require.NoError(t, err)
+
+	operatorName := fmt.Sprintf("operator-%s", rand.String(8))
+	operator := &operatorv1alpha1.Operator{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: operatorName,
+		},
+	}
+	return operator, operatorName, operatorCatalog
+}
+
+func testCleanup(t *testing.T, cat *catalogd.Catalog, operator *operatorv1alpha1.Operator) {
+	// getArtifactsOutout?
```

Comment:
Seems reasonable to place it here if all the tests end up calling this function. IIRC getting the artifacts should only happen for tests that have failed.

Reply:
I see further down it is just called directly. Would remove this comment if that is the route being taken

Reply:
I think the original test was a bit weird, in that there were multiple places that `getArtifactOutput` was called, and I cleaned it up to be run after every test, rather than at the end.